pyinfra/operations/git.py

```diff
@@ -245,6 +298,27 @@ def worktree(
 
         yield command
 
+    # It exists and we still want it => pull/rebase it
+    elif host.fact.directory(worktree) and present:
+
+        # pull the worktree only if it's already linked to a tracking branch or
+        # if a remote branch is set
+        if host.fact.git_tracking_branch(worktree) or from_remote_branch:
+            command = 'cd {0} && git pull'.format(worktree)
+
+            if rebase:
+                command += ' --rebase'
+
+            if from_remote_branch:
+                if '/' not in from_remote_branch:
+                    raise OperationError(
+                        'The remote branch must follow the following format REMOTE/BRANCH_NAME '
+                        '(for example: origin/master)',
+                    )
```

Comment:
It's fragile. Branches can contain `/`. You may even have a local branch named "origin/master".

To avoid ambiguities, I would change the API and force the `from_remote_branch` argument to be a 2-tuple `(remote, branch)`